ts/features/services/screens/ServiceDetailsScreen.tsx

```diff
@@ -60,24 +65,36 @@ export const ServiceDetailsScreen = ({ route }: ServiceDetailsScreenProps) => {
 
   const dispatch = useIODispatch();
 
-  const servicePot = useIOSelector(state =>
-    serviceByIdSelector(state, serviceId)
-  );
-
   useEffect(() => {
     dispatch(loadServiceDetail.request(serviceId));
+    dispatch(loadServicePreference.request(serviceId));
   }, [dispatch, serviceId]);
 
-  if (pot.isError(servicePot)) {
+  const serviceById = useIOSelector(state =>
+    serviceByIdSelector(state, serviceId)
+  );
+
+  const isLoadingServiceById = useIOSelector(state =>
+    isLoadingServiceByIdSelector(state, serviceId)
+  );
+
+  const isErrorServiceById = useIOSelector(state =>
+    isErrorServiceByIdSelector(state, serviceId)
+  );
+
+  if (!serviceById) {
+    return null;
+  }
+
+  if (isErrorServiceById) {
     return <ServiceDetailsFailure serviceId={serviceId} />;
   }
 
-  if (pot.isLoading(servicePot) || pot.isNone(servicePot)) {
+  if (isLoadingServiceById) {
```

Comment:
Are we sure that something will fit here?

Reply:
yes (good observation), this PR changes the behavior: the skeleton will be displayed if the pot is in the loading state.